### ISSUE TITLE
Unexpected results with roundToNearstMinutes

### ISSUE THREAD

**Issue by pascalporedda** [date-fns/date-fns#3242]
Hey guys,

thanks for this great library, I have been using it for quite a while. Now I have discovered a bug with the roundToNearestMinutes function.

I wrote some unit test cases that should explain the bug quite clearly, one might use to verify it is working as expected:

```ts

  it('should round a date up and down as expected', () => {
    const now = setMinutes(new Date(), 28);

    let rounded = roundToNearestMinutes(now, {
      nearestTo: 5,
      roundingMethod: 'floor',
    });

    expect(getMinutes(rounded)).toBe(25); // right now it returns 30

    rounded = roundToNearestMinutes(now, {
      nearestTo: 5,
      roundingMethod: 'ceil',
    });

    expect(getMinutes(rounded)).toBe(30); // right now it returns 35
});
```

Edit: I'm using v2.29.3

**Comment by fturmel** [comment]
#3132 will fix this issue, but still waiting for a maintainer to comment

**Comment by fturmel** [comment]
Fixed in v3.2.0!
